main/src/ca/uwaterloo/flix/language/phase/Reducer.scala

```diff
@@ -94,7 +94,8 @@ object Reducer {
         Expr.ApplyClo(e1, e2, ct, tpe, purity, loc)
 
       case Expr.ApplyDef(sym, exps, ct, tpe, purity, loc) =>
-        if (ct == ExpPosition.NonTail && Purity.isControlImpure(purity)) lctx.pcPoints += 1
```

Comment:
@magnus-madsen the current code looks at the whole effect of the function call, i.e. the effects of the functions __and__  the effects of the arguments. This PR is changing it to look at the actual function effect. The current code thinks more things are control-impure than there actually are